### DISCUSSION TITLE
Server and client auth state not in sync for Google OAuth

### DISCUSSION THREAD

**Discussion by AdrianKrebs** [supabase/supabase#12074] [Questions] [answered]
I have a login with email or google OAuth. After a successful login I'm redirecting to a protected route.
Everything works fine for the email/password login, but with Google, the app redirects to the protected route after the successful OAuth login and can't get a session in `getServerSideProps`. It almost looks like a timing issue since it's sometimes working and I'm seeing a session on the client side.

Any ideas?

``` 
export const getServerSideProps = async (ctx) => {
const supabase = createServerSupabaseClient(ctx)
  // Check if we have a session
  const {
    data: { session },
  } = await supabase.auth.getSession()
  if (!session) {
    console.log('no session')
    return {
      redirect: {
        destination: '/signin',
        permanent: false,
      },
    }
  }

  return {
    props: {
      initialSession: session,
      user: session.user
    },
  }
```

**Comment by GaryAustin1** [comment, marked as answer]
I believe the redirect URL needs to be under control of the browser client.  Not absolutely clear from your comment where the redirect URL goes to.  You can't redirect to a protected route because the browser has to confirm the token in the URL and set the client up.
https://github.com/supabase/auth-helpers/issues/391 discusses the issue (but for a middleware page)
And I apologize if this not your case.

**Reply by AdrianKrebs** [reply]
That was it, thanks!
